src/WamNode.js

```diff
@@ -248,6 +248,8 @@ export default class WamNode extends AudioWorkletNode {
 			if (ids.length) {
 				this._pendingResponses[id] = resolve;
 				this.port.postMessage({ id, request });
+			} else {
```

Comment:
i’m not sure what’s preferable from a ‘best practices’ standpoint, but another option would be to do something like this right?
```
async clearEvents() {
  const request = 'remove/events';
  const id = this._generateMessageId();
  const ids = Object.keys(this._pendingEvents);
  if (ids.length) {
    return new Promise((resolve) => {
      this._pendingResponses[id] = resolve;
      this.port.postMessage({ id, request });
    }).then((clearedIds) => {
      clearedIds.forEach((clearedId) => {
        this._pendingEvents[clearedId]();
        delete this._pendingEvents[clearedId];
      });
    });
  }
}
```
seems a little cleaner to me

Reply:
changed to this one, better than my quick fix ;)